client/my-sites/checkout/composite-checkout/hooks/use-prepare-products-for-cart.ts

```diff
@@ -183,7 +186,7 @@ function chooseAddHandler( {
 		return 'doNotAdd';
 	}
 
-	if ( isLoggedOutCart || isNoSiteCart ) {
+	if ( ( ! isGiftPurchase && isLoggedOutCart ) || isNoSiteCart ) {
```

Comment:
What's the reason for not using `addFromLocalStorage` for gifting?

Reply:
As the gift is actually a subscription renewal, we have to manage it as such, otherwise, the product is invalid. (I understand that what it does is validate the subscription + product)

Reply:
Interesting, thanks!

Reply:
It might be worth adding this explanation in a comment here since it's not obvious (honestly it wasn't obvious before but we've got to start somewhere 😅 ).

Reply:
Done!